tempodb/encoding/vparquet4/block_autocomplete.go

```diff
@@ -114,6 +128,128 @@ func createDistinctIterator(
 	return currentIter, nil
 }
 
+func createDistinctEventIterator(
+	makeIter makeIterFn,
+	tag traceql.Attribute,
+	primaryIter parquetquery.Iterator,
+	conditions []traceql.Condition,
+) (parquetquery.Iterator, error) {
+	var (
+		columnSelectAs    = map[string]string{}
+		iters             []parquetquery.Iterator
+		genericConditions []traceql.Condition
+	)
+
+	// TODO: Potentially problematic when wanted attribute is also part of a condition
+	//     e.g. { span.foo =~ ".*" && span.foo = }
+	addSelectAs := func(attr traceql.Attribute, columnPath, selectAs string) {
+		if attr == tag {
+			columnSelectAs[columnPath] = selectAs
+		} else {
+			columnSelectAs[columnPath] = "" // Don't select, just filter
+		}
+	}
+
+	for _, cond := range conditions {
+		// Intrinsic?
+		switch cond.Attribute.Intrinsic {
+		case traceql.IntrinsicEventName:
+			pred, err := createStringPredicate(cond.Op, cond.Operands)
+			if err != nil {
+				return nil, err
+			}
+			iters = append(iters, makeIter(columnPathEventName, pred, columnSelectAs[columnPathEventName]))
+			addSelectAs(cond.Attribute, columnPathEventName, columnPathEventName)
+			continue
+		}
+		// Else: generic attribute lookup
+		genericConditions = append(genericConditions, cond)
+	}
+
+	attrIter, err := createDistinctAttributeIterator(makeIter, tag, genericConditions, DefinitionLevelResourceSpansILSSpanEventAttrs,
+		columnPathEventAttrKey, columnPathEventAttrString, columnPathEventAttrInt, columnPathEventAttrDouble, columnPathEventAttrBool)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating event attribute iterator")
+	}
+
+	// if no intrinsics and no primary then we can just return the attribute iterator
+	// if len(iters) == 0 && primaryIter == nil {
+	// 	return attrIter, nil
+	// }
```

Comment:
Is this meant to be commented out?  Just asking because the equivalent condition in `createDistinctLinkIterator` is not commented out (see L236)